docs/minimal/transformations.md

```diff
@@ -0,0 +1,53 @@
+4. If any custom parameters are needed in the custom transformation logic, they can be passed via the *customParameters* input to the template. These parameters will be passed to the *init* method of the custom class. The *init* method is invoked once per worker setup.
+
+Implementation details for custom transformation:
+1. [MigrationTransformationRequest](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/spanner-migrations-sdk/src/main/java/com/google/cloud/teleport/v2/spanner/utils/MigrationTransformationRequest.java) contains the following information - 
+    - tableName - Name of the source table to which the event belongs to.
+    - shardId - Logical shard id of the record.
+    - eventType - The event type can either be INSERT, UPDATE-INSERT, UPDATE, UPDATE_DELETE or DELETE. Please refer to the [datastream documentation](https://cloud.google.com/datastream/docs/events-and-streams) for more details.
+    - requestRow - It is a map where key is the source column name and value is source column value.
+2. [MigrationTransformationResponse](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/spanner-migrations-sdk/src/main/java/com/google/cloud/teleport/v2/spanner/utils/MigrationTransformationResponse.java) contains the following information - 
+    - responseRow - It is a map where key is the spanner column name and value is spanner column value.
+    - isEventFiltered - If set to true, event will be skipped and not written to spanner.
+3. Values in the response row should be of same datatype as the spanner schema.
+4. Please throw **InvalidTransformationException** in case of any error while processing a particular event in custom jar.
+
+Here is a table that details the source data type for **MySQL**, its corresponding request row object type, spanner datatype and the expected response format:
+| Source datatype         | Request object type                 | Spanner datatype      | Response format                                                                        |
+|-------------------------|-------------------------------------|-----------------------|----------------------------------------------------------------------------------------|
+| TINYINT                 | Long                              | INT64     | Long                                                                                 |
+| INT                     | Long                              | INT64     | Long                                                                                 |
+| BIGINT                  | Long                              | INT64    | Long                                                                                 |
+| TIME                    | Long ([time-micros](https://avro.apache.org/docs/current/specification/_print/#time-microsecond-precision) ex: 45296000000 for 12:34:56) | STRING      | Long|
```

Comment:
Note: b/349257952 tracks this which would cause this mapping to change, for both source and destination.

Reply:
Will update the documentation once the bug is fixed